api/src/main/java/org/apache/iceberg/transforms/Dates.java

```diff
@@ -75,6 +80,10 @@ public Type getResultType(Type sourceType) {
       return Expressions.predicate(pred.op(), fieldName);
     } else if (pred.isLiteralPredicate()) {
       return ProjectionUtil.truncateInteger(fieldName, pred.asLiteralPredicate(), this);
+    } else if (pred.isSetPredicate() && pred.op() == IN) {
```

Comment:
This looks the same as the `Bucket` implementation. Can we refactor this code to use a common method, like the `ProjectionUtil.truncateInteger` used above? How about `ProjectionUtil.transformSet(String, BoundSetPredicate, Transform)`?

Reply:
Yeah, you are right. Refactored the code and added `transformSet` static method.